streaming/src/main/scala/org/apache/spark/streaming/ui/UIUtils.scala

```diff
@@ -67,9 +67,9 @@ private[streaming] object UIUtils {
    * will discard the fractional part.
    */
   def convertToTimeUnit(milliseconds: Long, unit: TimeUnit): Double = unit match {
-    case TimeUnit.NANOSECONDS => milliseconds * 1000 * 1000
-    case TimeUnit.MICROSECONDS => milliseconds * 1000
-    case TimeUnit.MILLISECONDS => milliseconds
+    case TimeUnit.NANOSECONDS => milliseconds * 1000.0 * 1000.0
```

Comment:
How about unifying it to `milliseconds.toDouble`?

Reply:
Done! Let me know if my change isn't what you meant